SQS -> Pulsar topic routing enhancement

seburke opened the pull request:
Added functionality for routing to many topics based on a SQS message attribute

freeznet commented on src/main/java/org/apache/pulsar/ecosystem/io/sqs/SQSRecord.java:
```diff
@@ -97,6 +97,10 @@ public void fail() {
     public Optional<String> getDestinationTopic() {
-        return Optional.of(destination);
+        if (msg.getMessageAttributes().containsKey(SQSUtils.PULSAR_TOPIC_ATTRIBUTE)) {
+            return Optional.of(msg.getMessageAttributes().get(SQSUtils.PULSAR_TOPIC_ATTRIBUTE).getStringValue());
```
Can we have some validation with the topic name?

Anonymitaet commented:
@freeznet thanks for your info. Now, @Huanli-Meng is the owner of connectors, she can help handle connector docs.

seburke commented:
Just circling back with this if there is anything else needed to get this merged. 
Thanks!

freeznet reviewed (approved):
LGTM

freeznet commented:
@nlu90 could you please help to review this PR when you have time, thanks.

nlu90 reviewed (changes requested):
I'm a little concerned introducing the google guava and apache commons dependency for implementing one method.

Can we do it with the native java language features for now and avoid introducing such dependencies?

seburke commented:
> I'm a little concerned introducing the google guava and apache commons dependency for implementing one method.
> 
> Can we do it with the native java language features for now and avoid introducing such dependencies?

I removed the new dependencies as you mentioned. Getting rid of the checkArgument method was straightforward. The isNotBlank method is a little more complex and seems unnecessary to implement when the awssdk has the same method and the implementations are pretty much the same so I switched back to that import since it is already imported anyway. If you want something else, happy to go that route, this just seemed easiest and I was trying to copy what the full pulsar topic validation is as closely as possible.

nlu90 reviewed (approved):
LGTM

nlu90 commented:
> > I'm a little concerned introducing the google guava and apache commons dependency for implementing one method.
> > Can we do it with the native java language features for now and avoid introducing such dependencies?
> 
> I removed the new dependencies as you mentioned. Getting rid of the checkArgument method was straightforward. The isNotBlank method is a little more complex and seems unnecessary to implement when the awssdk has the same method and the implementations are pretty much the same so I switched back to that import since it is already imported anyway. If you want something else, happy to go that route, this just seemed easiest and I was trying to copy what the full pulsar topic validation is as closely as possible.

Thanks for your contribution!

freeznet commented:
@seburke would you please rebase the master so that the integration tests CI can be fixed, thanks.
